### PR TITLE
refactor(@angular/cli): removed the github pages deploy command

### PR DESCRIPTION
**Pull request by Brocco**
BREAKING CHANGE: This command is being removed from the core of the CLI
There are several options for deploying CLI-based applications outside the scope of this project.
One of which being https://github.com/angular-buch/angular-cli-ghpages
This functionality is likely to return in the form of an addon/plugin in the future

### CODE REVIEW

File: packages/@angular/cli/utilities/completion.sh [hunk 1]
```diff
@@ -6,11 +6,10 @@
 #           or  ng completion 1>> ~/.zshrc 2>>&1
 #
 
-ng_opts='b build completion doc e2e g generate get github-pages:deploy gh-pages:deploy h help i init install lint make-this-awesome new s serve server set t test v version'
+ng_opts='b build completion doc e2e g generate get h help i init install lint make-this-awesome new s serve server set t test v version'
```
Comment (hansl) [inline packages/@angular/cli/utilities/completion.sh]: Could you undo the changes to completion.sh? We have a change coming to it and this won't be needed, and I'd like to commit both without having to rebase one of you ;)

**Comment by Brocco** [comment]
nit addressed

**Comment by hansl** [comment]
LGTM.

**Comment by JohannesHoppe** [comment]
Thanks for mentioning us. I expect some more traffic now! :smile:

**Comment by filipesilva** [comment]
@JohannesHoppe I always thought your command was very well designed so you deserve it 👍

**Comment by petersgiles** [comment]
for anyone who needs more explicit instructions 

1) `npm install -g angular-cli-ghpages` (doesn't work if its not global)
2) Then in your package.json
`
  "scripts": {
    "deploy": "ng build -sm -ec -bh /repo name/ & ngh --silent=false",
`...
3) when you want to deploy `npm run deploy`

**Comment by JohannesHoppe** [comment]
Don't forget to copy the index.html to 404.html! ;-)

**Comment by JohannesHoppe** [comment]
@petersgiles Could you please verify the the `ngh` command really just works globally? It's supposed to work local (`npm i angular-cli-ghpages --save-dev` + run script in package.json) AND global (as described). 

If this is really the case, please submit an [issue](https://github.com/angular-buch/angular-cli-ghpages/issues).

**Comment by petersgiles** [comment]
@JohannesHoppe yep that's the way it happened for me.

I tried installing `npm install --save-dev angular-cli-ghpages` and it didn't make the `ngh` available so I uninstalled it and did `npm install -g angular-cli-ghpages` and it worked.

macOS Sierra

node: 6.9.5
os: darwin x64
@angular/common: 2.4.7
@angular/compiler: 2.4.7
@angular/core: 2.4.7
@angular/flex-layout: 2.0.0-beta.5
@angular/forms: 2.4.7
@angular/http: 2.4.7
@angular/material: 2.0.0-beta.1
@angular/platform-browser: 2.4.7
@angular/platform-browser-dynamic: 2.4.7
@angular/router: 3.4.7
@angular/cli: 1.0.0-beta.31
@angular/compiler-cli: 2.4.7

**Comment by JohannesHoppe** [comment]
@petersgiles I will take a look: https://github.com/angular-buch/angular-cli-ghpages/issues/5

**Comment by tkow** [comment]
It seems  @petersgiles 's problem not that ngh command is not incleded in your path because node install  packages in node_moudules/.bin when local installing. `$(pwd)/node_moudules/.bin` path are exported when npm commands  run before executed so that node's package commands are available in scripts  in package.json but not shell environment with no explicitly path prefix . npm support `npm bin` command which get path `$(pwd)/node_modules/.bin` So, you can use ngh commands by `$(npm bin)/ngh ...`  . However, I have another problem catch message says "An error occurred!" in node v.7.4.0.
and details bellow.

```
(node:6042) UnhandledPromiseRejectionWarning: Unhandled promise rejection (rejection id: 2): Error: Unspecified error (run without silent option for detail)
(node:6042) DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
```
Please tell me how it cause to address, if there are someone feel not uncomfortable.I'll try to downgrade node version.Then , I move to angular-buch/angular-cli-ghpages repository for this ploblem is not angular-cli.

**Comment by gkucmierz** [comment]
Why it was removed from ng-cli?

It was working good here, but now this external package is broken
https://github.com/angular-buch/angular-cli-ghpages/issues/9

**Comment by JohannesHoppe** [comment]
@tkow - please see
https://github.com/angular-buch/angular-cli-ghpages/issues/5 
https://github.com/angular-buch/angular-cli-ghpages/issues/11
for further discusson

**Comment by JohannesHoppe** [comment]
@gkucmierz You can always deploy your `/dist` folder "by hand", it you don't like `angular-cli-ghpages`!
Some instructions are [here](https://github.com/weavejester/codox/wiki/Deploying-to-GitHub-Pages) (replace "docs" by "dist).

**Comment by Brocco** [comment]
@gkucmierz there were many reported issues with this command, so rather than ship the CLI with the issues it was decided to remove it, especially with some viable community alternatives.

**Comment by urish** [comment]
An alternative is the [gh-pages](https://www.npmjs.com/package/gh-pages) package, I use it for ages and it works fine:

    npm install --save-dev gh-pages

then, in your package.json:

    "scripts": {
        "deploy": "ng build -prod -sm -ec -bh /reponame/ && gh-pages -d dist"
    }

(don't forget to replace `/reponame/` with your actual repo name)

**Comment by JohannesHoppe** [comment]
Unfortionally this does not work out of the box on Travis-CI. That was the main reason to start `angular-cli-ghpages` on top of `ghpages`.

**Comment by aakash14goplani** [comment]
@petersgiles 
In your explicit instructions `"scripts": { "deploy": "ng build -sm -ec -bh /repo name/ & ngh --silent=false",` throws error 
> error: unknown option `--silent'

thing that work for me was `ngh --no-silent`

**Comment by Brocco** [comment]
@aakash14goplani Can you please take your current issue over the https://github.com/angular-buch/angular-cli-ghpages repo?

Thanks! And good luck getting your app deployed.

**Comment by JohannesHoppe** [comment]
Hello, friends. In Angular CLI 8.3.0-next.0. the builders where enhanced which gives us the command: `ng deploy`

This means that the old "github pages deploy command" also returns to the Angular CLI:

```sh
ng update @angular/cli @angular/core --next=true
ng add angular-cli-ghpages
ng deploy
```

angular-cli-ghpages-deploy

More at: 
https://github.com/angular-schule/angular-cli-ghpages

Wishing you a wonderful day! 😃

**Comment by filipesilva** [comment]
@JohannesHoppe that's pretty nifty! cc @mgechev
